Tasks/InstallSSHKey/installsshkey-util.ts

```diff
@@ -0,0 +1,118 @@
+                    tl.setVariable(key, value);
+                    tl.setTaskVariable(postKillAgentSetting, value);    
+                }
+                else if (sshAgentSockEnvVariableKey === key) {
+                    tl.setVariable(key, value);
+                }
+            } else {
+                tl.debug('Skipping ' + elements[i]);
+            }
+        }
+    }
+
+    public installKey(publicKey: string, privateKeyLocation: string) {
+        tl.debug('Get a list of the SSH keys in the agent');
+        let results: trm.IExecSyncResult = tl.execSync(this.getExecutable('ssh-add'), '-L');
+
+        let publicKeyComponents:string[] = publicKey.split(' ');
+        if (publicKeyComponents.length <= 1) {
+            throw tl.loc('SSHPublicKeyMalformed');
+        }
+
+        let publicKeyHash: string = publicKeyComponents[1];
+        tl.debug('Checking for public SSH key: ' + publicKeyHash);
+        if (results.stdout.indexOf(publicKeyHash) !== -1) {
+            throw tl.loc('SSHKeyAlreadyInstalled');
+        }
+
+        tl.debug('Adding the SSH key to the agent ' + privateKeyLocation);
+        results = tl.execSync(this.getExecutable('ssh-add'), privateKeyLocation);
+        if (results.error) {
+            throw tl.loc('SSHKeyInstallFailed');
+        }
+        tl.setTaskVariable(postDeleteKeySetting, privateKeyLocation);
+
+        results = tl.execSync(this.getExecutable('ssh-add'), null);
+    }
+
+    public deleteKey(key: string) {
+        let deleteKey: string = tl.getTaskVariable(postDeleteKeySetting);
+        if (deleteKey) {
+            tl.debug('Deleting Key: ' + deleteKey);
+            tl.execSync(path.join(external, 'ssh-add.exe'), '-d ' + deleteKey);
+        }
+    }
+}
+
+export function setKnownHosts(knownHostsEntry: string) {
+    let knownHostsFolder: string = path.join(os.homedir(), '.ssh');
+    if (!fs.existsSync(knownHostsFolder)) {
+        fs.mkdirSync(knownHostsFolder);
+    }
+    let knownHostsFile: string = path.join(knownHostsFolder, 'known_hosts');
+
+    tl.debug('Read known_hosts');
+    tl.setTaskVariable(postKnownHostsContentsSetting, fs.readFileSync(knownHostsFile).toString());
+    tl.setTaskVariable(postKnownHostsLocationSetting, knownHostsFile);
+
+    tl.debug('Inserting entry into known_hosts');
+    fs.writeFileSync(knownHostsFile, knownHostsEntry + os.EOL);
+}
+
+export function tryRestoreKnownHosts() {
+    let knownHostsContents: string = tl.getTaskVariable(postKnownHostsContentsSetting);
+    let knownHostsLocation: string = tl.getTaskVariable(postKnownHostsLocationSetting);
+
+    tl.debug('Restoring known_hosts');
+    if (knownHostsContents && knownHostsLocation) {
+        fs.writeFileSync(knownHostsLocation, knownHostsContents);
+    } else if (knownHostsLocation || knownHostsContents) {
+        tl.warning('Inconsistency with known_hosts cannot reset (location=' + knownHostsLocation + ' content=' + knownHostsContents + ')');
```

Comment:
This warning should be localized and is a little difficult to understand.  Or did you mean for it to be debug output?